File: server/ruby/app/infrastructure/mappers/payment_details/card_essentials_mapper.rb
module Infrastructure
  module Mappers
    module PaymentDetails
      class CardEssentialsMapper
        def self.from_sdk_response(response)
          return nil if response.nil?

          dto = Business::Domain::Payments::PaymentDetails::CardEssentials.new
          dto.country_code = response.country_code
          dto.card_number = response.card_number
          dto.expiry_date = response.expiry_date
          dto.bin = response.bin

          dto
        end
      end
    end
  end
end

File: server/python/app/infrastructure/mappers/payment_details/card_essentials_mapper.py
from typing import Optional
from app.application.domain.payments.payment_details.card_essentials import CardEssentials
from onlinepayments.sdk.domain.card_essentials import CardEssentials as CardEssentialsSdk

class CardEssentialsMapper:

    @staticmethod
    def map_from_sdk_response(response: Optional[CardEssentialsSdk]) -> Optional[CardEssentials]:
        if response is None:
            return None

        dto = CardEssentials()
        dto.country_code = response.country_code
        dto.card_number = response.card_number
        dto.expiry_date = response.expiry_date
        dto.bin = response.bin

        return dto